addons/mass_mailing/controllers/portal.py

```diff
@@ -0,0 +1,18 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from werkzeug.exceptions import NotFound
+
+from odoo.http import Controller, request, route
+
+
+class MassMailingController(Controller):
+
+    @route('/mass_mailing/turnoff_report', type='http', website=True, auth='public')
```

Comment:
Security issue 🔥

I can send you a link to a web page that contains an image with this URL as the source, and it will deactivate the feature without you knowing what happened (that why we use CSRF token in the website forms). So I think it's worth it to generate a token.

(note: the image is the most sniky way to do it, but it should be on the same website I think, but I can also just redirect you on the page)

Reply:
shoudln't it be `auth="user"` ?